VideoLocker/src/main/java/org/edx/mobile/view/CourseDashboardFragment.java

```diff
@@ -30,111 +28,124 @@
     static public String CourseData = TAG + ".course_data";
 
     private EnrolledCoursesResponse courseData;
+    private boolean IS_COURSEWARE_ACCESSIBLE = true;
 
     @Inject
     IEdxEnvironment environment;
-    @InjectView(R.id.course_detail_name)
     private TextView courseTextName;
-    @InjectView(R.id.course_detail_extras)
     private TextView courseTextDetails;
-    @InjectView(R.id.header_image_view)
     private ImageView headerImageView;
-    @InjectView(R.id.dashboard_detail)
     private LinearLayout parent;
+    private TextView errorText;
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        final Bundle bundle = getArguments();
+        courseData = (EnrolledCoursesResponse) bundle.getSerializable(CourseData);
+        if (courseData != null) {
+            IS_COURSEWARE_ACCESSIBLE = courseData.getCourse().getCoursewareAccess().hasAccess();
+        }
     }
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.fragment_course_dashboard, container, false);
+        View view;
+        if (IS_COURSEWARE_ACCESSIBLE) {
+            view = inflater.inflate(R.layout.fragment_course_dashboard, container, false);
+            courseTextName = (TextView) view.findViewById(R.id.course_detail_name);
+            courseTextDetails = (TextView) view.findViewById(R.id.course_detail_extras);
+            headerImageView = (ImageView) view.findViewById(R.id.header_image_view);
+            parent = (LinearLayout) view.findViewById(R.id.dashboard_detail);
+        }
+        else {
```

Comment:
The `else` can be moved up to the same line as block ending. Avoids wasting space :wink: